var/lib/delphix-platform/os-migration

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+#
+# Copyright 2018 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+MIGRATE_CONFIG_SCRIPT="/opt/delphix/migration/migrate_config.py"
+MIGRATE_CONFIG_LOG="/var/delphix/migration/log"
+
+function die() {
+	echo "$(basename "$0"): $*" >&2
+	exit 1
+}
+
+function perform_migration() {
+	zpool import -f domain0 || die "Failed to import domain0"
+	# domain0/mds is owned by delphix:staff on Illumos and staff has GID 10.
+	# To minimize the changes in case we need to perform rollback, we add
+	# the appropriate group to the postgres user instead of changing the
+	# permissions of domain0/mds.
+	usermod -a -G 10 postgres ||
+		die "Failed adding group 10 to postgres user"
+	echo "$(date): Running migrate_config post-upgrade" \
+		>>"$MIGRATE_CONFIG_LOG"
+	"$MIGRATE_CONFIG_SCRIPT" post-upgrade >>"$MIGRATE_CONFIG_LOG" ||
+		die "Failed migrate_config post-upgrade"
+	rm /os-migration || die "Failed to remove /os-migration"
+}
+
+if [[ -f /os-migration ]]; then
```

Comment:
It'd be good to comment on where this file (i.e. global variable/state) comes from, since it's not clear looking at this script in isolation.

Additionally, would it be better to store this in something like `/var/lib/delphix-migration/whatever`? rather than polluting `/`?

Upgrade also uses global variables/state, and stores it as ZFS dataset properties; so that's another option. i.e. it sets a property, then clears it when upgrade is "done"; very similar to what you're doing.

Reply:
I could use `/var/delphix/migration/` instead, since I already use it for other purposes. The idea of using `/` came from the `/reconfigure` flag that was used in Illumos, but `/var/delphix/migration/` definitely seems like a better place in this case.